apps/chat/channels.py

```diff
@@ -895,6 +895,67 @@ def echo_transcript(self, transcript: str):
             self.participant_identifier, text=f"I heard: {transcript}", reply_to_message_id=self.message.message_id
         )
 
+    @property
+    def supports_multimedia(self) -> bool:
+        return True
+
+    def _get_supported_unsupported_files(self, files: list[File]) -> tuple[list[File], list[File]]:
+        supported_files = []
+        unsupported_files = []
+
+        for file in files:
+            mime = file.content_type
+            size = file.content_size or 0  # in bytes
+
+            is_inline_supported = False
+            if (
+                mime.startswith("image/")
+                and size <= 10 * 1024 * 1024  # 10mb
+                or mime.startswith(("video/", "audio/", "application/"))
+                and size <= 50 * 1024 * 1024  # 50mb
+            ):
+                is_inline_supported = True
+
+            if is_inline_supported:
+                supported_files.append(file)
+            else:
+                unsupported_files.append(file)
+
+        return supported_files, unsupported_files
+
+    def send_file_to_user(self, file: File):
+        chat_id = self.participant_identifier
+        mime = file.content_type
+        file_data = file.file
+
+        # Mapping of MIME type prefixes to Telegram methods and argument names
+        mime_method_map = {
+            "image/": ("send_photo", "photo"),
+            "video/": ("send_video", "video"),
+            "audio/": ("send_audio", "audio"),
+        }
+
+        for prefix, (method_name, arg_name) in mime_method_map.items():
```

Comment:
To prevent having to always loop through the `mime_method_map` and maybe use one of the methods in there, I'd suggest taking another approach. Maybe something like this:

```python
method = None
arg_name = ""
match mime.split("/")[0]:
    case "image":
        method = self.telegram_bot.send_photo
        arg_name = "photo"
    case "video":
        method = self.telegram_bot.send_video
        arg_name = "video"
    case "audio":
        method = self.telegram_bot.send_audio
        arg_name = "audio"
    case _:
        method = self.telegram_bot.send_document
        arg_name = "document"

antiflood(method, chat_id, **{arg_name: file_data})
```